examples/jwt-client-authn.js

```diff
@@ -13,9 +13,12 @@ app.use(
       response_type: 'code',
     },
     clientAuthMethod: 'private_key_jwt',
-    clientAssertionSigningKey: fs.readFileSync(
+    clientAssertionConfig: {
```

Comment:
We don't need to specify the alg if we're not publishing a jwks uri

Reply:
@adamjmcgrath - There are some cases where you need to explicitly provide an alg. For instance you can use an RSA private key with any of RS 256/384/512, and somewhere along the way openid-client or jose are defaulting to RS256 for an RSA private key.

I can think of two options to let callers specify their alg:
1. Keep the current setup, noting that we're passing the alg to openid-client [here](https://github.com/auth0/express-openid-connect/pull/291/files#diff-ca407d959d33ce92a7f7efee354a5ea5e6e61fcd797d32c04205e429a074ed54:~:text=%3F%20%7B-,keys%3A%20%5B,%5D%2C,-%7D) after parsing with jose's JWK module
2. Explicitly require a jwk and inspect the alg (maybe warning if it's missing and that we'll choose a default)

Note that openid-clilent won't throw an error for invalid algs until runtime. For example, if I try to specify ES384 but give an RSA private key I get the following, but not until I actually attempt a grant:
```
BadRequestError: no key found in client jwks to sign a client assertion with using alg ES384
    at C:\Users\adams\source\node\express-openid-connect\middleware\auth.js:138:31
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

What do you think?

Reply:
Hey @madaster97 - sorry for the delay responding, I was away last week

Your right, we should allow specifying the algorithm, but it should be optional

By default `openid-client` uses one of the `token_endpoint_auth_signing_alg_values_supported` if you don't specify - so I just need to figure out whether the default should be `RS256` or `undefined` (let openid-client pick one from `token_endpoint_auth_signing_alg_values_supported`)

Also, can we make the config flat,eg `clientAssertionSigningKey` and `clientAssertionSigningKeyAlg` (like `idTokenSigningAlg`)